src/core/thread/mle.hpp

```diff
@@ -718,6 +719,36 @@ class Mle : public InstanceLocator, private NonCopyable
 
 #endif // OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
 
+#if OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE
+    /**
+     * Indicates whether a Wake-up End Device is being attached.
+     *
+     * @retval true   A Wake-up End Device is being attached.
+     * @retval false  A Wake-up End Device is not being attached.
+     */
+    bool IsWedAttaching(void) const { return mWedAttachState == kWedAttaching; }
+
+    /**
+     * Indicates whether a Wake-up End Device is currently attached.
+     *
+     * @retval true   A Wake-up End Device is attached.
+     * @retval false  A Wake-up End Device is not attached.
+     */
+    bool IsWedAttached(void) const { return mWedAttachState == kWedAttached || mWedAttachState == kWedDetaching; }
```

Comment:
Because technically, the device is still attached then. But ok, I can for now remove this code until it's used so it's clearer why this approach is taken.